keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -55,6 +55,14 @@ class ApplicationService(
 
   fun getDeliveryConfig(application: String) = repository.getDeliveryConfigForApplication(application)
 
+  fun deleteConfigByApp(application: String) =
+    try {
+      val config = repository.getDeliveryConfigForApplication(application)
```

Comment:
I feel like the repository should be able to delete the delivery config based on the application name to avoid the extra query here.

Reply:
P.S. Not a blocker, though.

Reply:
fair point. I should just implement that lol

Reply:
HA! It exists. I shall use it. Oops.